webapp-django/crashstats/crashstats/management/commands/verifyprocessed.py

```diff
@@ -27,24 +28,42 @@
 RAW_CRASH_PREFIX_TEMPLATE = "v2/raw_crash/%s/%s/"
 PROCESSED_CRASH_TEMPLATE = "v1/processed_crash/%s"
 
+# Number of seconds until we decide a worker has stalled
+WORKER_TIMEOUT = 10 * 60
+
 
 metrics = markus.get_metrics("cron.verifyprocessed")
 
 
-def check_crashids(entropy, boto_conn, bucket_name, date):
+def check_crashids(entropy_chunk, date):
```

Comment:
The boto S3 client was pickleable, but the boto3 S3 client is not. The problem that creates is that when the main process spins off a bunch of subprocesses to do the work, it can no longer pass along a boto3 S3 client. Thus each subprocess has to create its own client.

Creating a client takes 0.5s for some reason. We used to create like 4096 processes. So that's another 2000 seconds or so. That makes the script take a *LOT* longer to run.

The other thing I noticed in the "the bucket is empty" case, is that this script spends an awful lot of time creating and handling processes. `verifyprocessed` takes 2m to run in the master branch with nothing in the bucket. That's kind of silly.

To reduce the cost of creating boto3 S3 clients and the overhead of creating and handling processes, I changed this to chunk entropy values. Now each process handles 5 entropy values rather than 1. That significantly reduces the work required for the "there's nothing in the bucket" case and it takes 1m30s or so for me to run.

Socorro is using Python 3.6. Once we get to 3.7, it might be even better to switch to asyncio since this is mostly waiting on network i/o.